src/app/lib/README.md

```diff
@@ -41,3 +41,92 @@ Here are the font sizes in `px`s:
+| 1280px +        | 16px    | 16px    | static / 1280px        |
+
+## Article layout requirements
+
+In the article project we have requirements for items to span wider than the grid, these "full bleed" items will span the full width of the viewport effectively breaking out of the grid.
+
+The article layout UX design has the following requirements:
+
+| Breakpoints     | Columns | Content* spans X columns | 
+| :-------------- | :------ | :----------------------- | 
+| 0px - 599px     | 6       | 6 of 6                   | 
+| 600px - 1007px  | 6       | 4 of 6                   | 
+| 1008px - 1280px | 8       | 6 of 8                   | 
+| 1280px +        | 10      | 6 of 10                  | 
+* content contained within the grid, therefore not "full bleed"
+
+## Our CSS grid implementation
+
+Because of the above requirements we have a complex implementation of the grid mainly to support the "full bleed" component requirement.
+
+To achieve this we have used CSS grid which allows us to define both static or responsive columns depending on the UX grid definition.
+
+CSS Grid has the ability to add "grid-gaps" which are rendered inclusive of the grid as gaps between the columns/rows. This means we can directly use them as the gutters to match the UX grid.
+
+Below 1008px we define all columns to be dynamic. 
+
+Above 1007px we statically define the widths of the columns due to the fact they are limited in the UX definition, which then allows us to have a dynamic column either side of the UX grid. 
+  EG: In CSS grid values we can specific `grid-template-columns: 1fr repeat(10, minmax(0, ${staticColWidth})) 1fr;` where the `1fr`s are dymanic and the columns and `repeat()` are the static width columns.
+
+The combination of the UX grid having margins prior to the column definitions and the product requirements for a "full bleed" component creates an interesting problem. We can't use CSS paddings or margins before the CSS grid implementation otherwise all components in the grid will be limited to the padding/margin and not able to go "full bleed". Our solution to this problem stems from the implementation of the static columns above 1007px - we can define a CSS grid that has an additional column either side of the UX defined columns so that the grid-gaps act both inbetween the columns as gutters and also outside of the grid as margins. 
+**NB: This implementation means at every viewport we have an additional column either side of the UX defined columns, often as a 0px column.**
+
+Therefore to meet all requirements we implement the following CSS grid where gutters and margins are grid-gaps:
+
+| Breakpoints     | Columns           | Gutters   | Margins          | Figure |
+| :-------------- | :---------------- | :-------- | :--------------- | :----- |
+| 0px - 399px     | 6 + 2 of 0px      | 0 of 8px  | 2 of 8px         | 1      |
+| 400px - 599px   | 6 + 2 of 8px*     | 5 of 8px  | 2 of 16px        | 2      |
```

Comment:
Don't think this needs an asterisk - you have explained three of the `Columns` rows below.

Do we need the Figure column? (I'd just remove)

Reply:
I think I was being overly explicit but agree it's overkill, will update